backend/BackendMIIT/src/main/java/com/example/BackendMIIT/service/impl/ProfileServiceImpl.java

```diff
@@ -41,14 +42,16 @@ public ProfileServiceImpl(ProfileRepository profileRepository,
     }
 
     @Override
-    public List<ProfileDto> getProfilesByInstituteName(String institute) {
-        List<Profile> profiles = profileRepository.findByInstitute(institute)
+    @Cacheable(value = "ProfileService::getProfilesByInstitute", key = "#name")
+    public List<ProfileDto> getProfilesByInstituteName(String name) {
```

Comment:
If you will decide to rename `instituteName` in the ProfileService.java, don't forget to rename this parameter

Reply:
Ok, thanks